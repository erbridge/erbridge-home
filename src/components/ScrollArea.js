import PropTypes from 'prop-types';
import Scrollbars from 'react-custom-scrollbars';

class ScrollArea extends Scrollbars {
  static childContextTypes = {
    scrollArea: PropTypes.object,
  };

  getChildContext() {
    return {
      scrollArea: {
        reset: () => {
          this.scrollToTop();
          this.scrollToLeft();
        },
      },
    };
  }
}

export default ScrollArea;
