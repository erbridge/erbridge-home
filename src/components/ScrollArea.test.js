import { shallow } from 'enzyme';
import React from 'react';

import ScrollArea from './ScrollArea';

it('renders correctly', () => {
  expect(shallow(<ScrollArea />)).toMatchSnapshot();
});
