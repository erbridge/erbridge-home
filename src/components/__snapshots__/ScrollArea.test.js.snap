// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "overflow": "hidden",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    key="view"
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "bottom": 0,
        "left": 0,
        "marginBottom": 0,
        "marginRight": 0,
        "overflow": "scroll",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <div
    key="trackHorizontal"
    style={
      Object {
        "borderRadius": 3,
        "bottom": 2,
        "display": "none",
        "height": 6,
        "left": 2,
        "position": "absolute",
        "right": 2,
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,.2)",
          "borderRadius": "inherit",
          "cursor": "pointer",
          "display": "block",
          "height": "100%",
          "position": "relative",
        }
      }
    />
  </div>
  <div
    key="trackVertical"
    style={
      Object {
        "borderRadius": 3,
        "bottom": 2,
        "display": "none",
        "position": "absolute",
        "right": 2,
        "top": 2,
        "width": 6,
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,.2)",
          "borderRadius": "inherit",
          "cursor": "pointer",
          "display": "block",
          "position": "relative",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
